src/test/java/support/fakes/FakeOkapi.java

```diff
@@ -97,6 +98,8 @@ private void registerFakeItemsStorageModule(Router router) {
       .withRootPath("/item-storage/items")
       .withRequiredProperties("materialTypeId", "permanentLoanTypeId")
       .withDefault("status", new JsonObject().put("name", "Available"))
+      .withCreateProcessors(new ItemCreateUpdateEffectiveLocationProcessor())
+      .withUpdateProcessors(new ItemCreateUpdateEffectiveLocationProcessor())
```

Comment:
I think I might be missing something with the processors model.

I understand the intent of this code is to take a record representation and process it in some way to produce a new record representation, in this case, add an `effectiveLocationId` property.

This could be achieved by using a Function<JsonObject, JsonObject> where the input is the current representation and the return value is the representation after processing.

I like the idea of describing this in an interface definition rather than a raw functional type. IIRC it is possible to define an instance of a single method interface implicitly using a method reference or similar. Could this be simplified to a single interface with an implicit implementation?